zapisy/apps/grade/poll/utils.py

```diff
@@ -156,21 +163,55 @@ def plot(self):
             used for embedding plots in the template.
         """
         if not self._components:
+            formatted_choices = [textwrap.fill(choice, 20) for choice in self._choices]
+
+            choices_with_non_zero_values = []
+            for i in range(len(self._choices)):
+                if self._choices_occurrences[i] != 0:
+                    choices_with_non_zero_values.append(formatted_choices[i])
+
+            answers_length = len(self._answers)
+            percents = []
+            text_values = []
+            if answers_length != 0:
+                for occurrences in self._choices_occurrences:
+                    percent = 100 * occurrences / answers_length
+                    percents.append(f"{percent:.1f} %".replace('.', ','))
+                    occurrences_str = str(occurrences)
+                    text_values.append("  "*(max(0, 3-len(occurrences_str)))+occurrences_str)
+
+            source_data = dict(choices=formatted_choices, values=self._choices_occurrences, percents=percents,
+                               text_values=text_values)
+
             plot = bokeh.plotting.figure(
-                y_range=self._choices,
+                y_range=formatted_choices,
                 sizing_mode='scale_width',
                 plot_height=250,
                 toolbar_location=None,
                 tools='',
+                min_border_left=130,
             )
 
-            source = bokeh.models.sources.ColumnDataSource(
-                data=dict(choices=self._choices, values=self._choices_occurences)
-            )
-
+            source = bokeh.models.sources.ColumnDataSource(data=source_data)
             plot.hbar(y='choices', right='values', source=source, height=0.8)
+
+            ticker_interval, last_tick = self._max_choice_occurrences.calculate_ticker_properties()
+
             plot.x_range.start = 0
-            plot.axis.minor_tick_line_color = None
+            plot.x_range.end = max(1, last_tick*1.1)
+            plot.xaxis.ticker = bokeh.models.tickers.SingleIntervalTicker(
+                interval=ticker_interval, num_minor_ticks=0
+            )
+
+            labels = LabelSet(y='choices', x='values', text='percents',
+                              x_offset=5, y_offset=-6, source=source, render_mode='canvas', text_font_size='11px')
+
+            labels2 = LabelSet(y='choices', x='values', text='text_values',
+                               x_offset=-22, y_offset=-4, source=source,
+                               render_mode='canvas', text_font_size='11px', text_color="white")
+
+            plot.add_layout(labels)
+            plot.add_layout(labels2)
```

Comment:
Etykiety rozjeżdżają się w pionie, przy czym to procenty są chyba lepiej wycentrowane, a nie liczby głosów. Z kolei procentom przydałoby się jakieś nieprzezroczyste tło (jeśli da się je łatwo nadać), żeby nie przeszkadzała pozioma linia, na której leżą. A tak naprawdę te poziome linie (w przeciwieństwie do pionowych) są w naszych wykresach tak naprawdę w ogóle niepotrzebne…

Plus: sam nie miałbym nic przeciwko nieco tasiemcowym wywołaniom `plot.add_layout(LabelSet(...))`, ale jeśli już decydujemy się nadać identyfikatory, to niech spełniają swoją funkcję (identyfikacyjną):
```suggestion
            percentage_labels = LabelSet(y='choices', x='values', text='percents',
                              x_offset=5, y_offset=-6, source=source, render_mode='canvas', text_font_size='11px')

            count_labels = LabelSet(y='choices', x='values', text='text_values',
                               x_offset=-22, y_offset=-6, source=source,
                               render_mode='canvas', text_font_size='11px', text_color="white")

            plot.add_layout(percentage_labels)
            plot.add_layout(count_labels)
```

Reply:
Usunęłam poziome linie, ale mam mieszane uczucia co do efektów. Zwłaszcza na mocno wypełnionym wykresie, mam wrażenie, że bez nich wygląda dziwnie:
![image](https://github.com/iiuni/projektzapisy/assets/25622641/efe37a8f-65d8-45d1-8186-a34920e0aaa4)
W wykresie pustszym, ich brak faktycznie raczje nie przeszkadza:
![image](https://github.com/iiuni/projektzapisy/assets/25622641/50f1354f-4d3c-44a5-b5c7-123ff4ec0bbc)